### DISCUSSION TITLE
dynamicLink().getInitialLink() Retrieves Old Link When Opening App Without Deeplink

### DISCUSSION THREAD

**Discussion by lmas3009** [invertase/react-native-firebase#7864] [Q&A]
I am experiencing an issue with Firebase Dynamic Link.

I have a link that opens the app via a deeplink using dynamicLink().getInitialLink(). I can see the post related to the link. However, when I open the app manually again without any link, I still see the same post that was opened via the deeplink. This issue occurs only on iOS.

On Android, there is no issue; I can see the post that was opened via the link and also the general posts when opened manually.

**Comment by idlework** [comment]
I have the issue. Subsequent calls will return the initial link. This behavior is different from the similar function `messaging().getInitialNotification()`. This function returns data on the first call, then on subsequent calls the function returns undefined.

Anyone know if this behaviour is expected? And if the difference between function is intentional?

**Reply by mikehardy** [reply]
Unfortunately I don't believe there will be any further development at all on dynamic links, the module is deprecated, will cease functioning in august, and the only efficient effort to spend on it is to migrate away from it with any available related time

https://firebase.google.com/support/dynamic-links-faq

**Reply by idlework** [reply]
Thanks for the quick answer @mikehardy. Too bad, but I was already anticipating an answer like this. I'm going to search / make a fix via a fork and then make sure it has the same behavior as `getInitialNotification`.
